### ISSUE TITLE
 why Assertion `contact_hdr != ((void *)0)' happened during registration process.

### ISSUE THREAD

**Issue by ysm0420** [pjsip/pjproject#3887]
### Describe the bug

Assertion happened  when state changed from Calling to Completed during register process

### Steps to reproduce

this issue happened many times in one day.  but  when i try to reproduce this issue, it did not happened any more.
from the below log, it can be found:
After a successful registration, for some unknown reason, TCP was disconnected suddently. In the following re-registration, an assertion occurred

### PJSIP version

2.12.1

### Context

platform:ARM926EJ-S rev 5 (v5l)
OS: Linux 4.4

### Log, call stack, etc

```shell
// register
05:38:43.510 tcpc0xb5959bcc  .TCP client transport created
05:38:43.511 tcpc0xb5959bcc  .TCP transport 10.42.93.8:40593 is connecting to 54.172.60.1:5060...
05:38:43.512   pjsua_core.c  .TX 689 bytes Request msg REGISTER/cseq=2051 (tdta0xb59666cc) to TCP 54.172.60.1:5060:
REGISTER sip:otiselevator-apac-stage.sip.twilio.com SIP/2.0
Via: SIP/2.0/TCP 10.42.93.8:40593;rport;branch=z9hG4bKPj5Lf4x.8V7RWblSxehhKV7Ptpu54EFBE8;alias
Route: <sip:otiselevator-apac-stage.sip.twilio.com;transport=tcp;lr>
Max-Forwards: 70
From: <sip:11265018@otiselevator-apac-stage.sip.twilio.com>;tag=HExLqlbt2b5gRXzskRGjTgQE4627fhmd
To: <sip:11265018@otiselevator-apac-stage.sip.twilio.com>
Call-ID: 9VesYyJIOdHCI-cZSFAnfh9SyCpMuntq
CSeq: 2051 REGISTER
Supported: outbound, path
Contact: <sip:11265018@122.97.221.234:2138;transport=TCP;ob>
Expires: 600
Allow: PRACK, INVITE, ACK, BYE, CANCEL, UPDATE, INFO, SUBSCRIBE, NOTIFY, REFER, MESSAGE, OPTIONS
Content-Length:  0


--end msg--
05:38:43.784 tcpc0xb5959bcc  TCP transport 10.42.93.8:40593 is connected to 54.172.60.1:5060
05:38:43.785            APP  SIP TCP transport is connected to 54.172.60.1:5060
05:38:44.076   pjsua_core.c  .RX 645 bytes Response msg 407/REGISTER/cseq=2051 (rdata0xb5959d7c) from TCP 54.172.60.1:5060:
SIP/2.0 407 Proxy Authentication required
CSeq: 2051 REGISTER
Call-ID: 9VesYyJIOdHCI-cZSFAnfh9SyCpMuntq
From: <sip:11265018@otiselevator-apac-stage.sip.twilio.com>;tag=HExLqlbt2b5gRXzskRGjTgQE4627fhmd
To: <sip:11265018@otiselevator-apac-stage.sip.twilio.com>;tag=67332571_c3356d0b_117a6b9e-b580-4c8f-b80a-1e9be7fa2187
Via: SIP/2.0/TCP 10.42.93.8:40593;received=112.80.12.234;rport=5447;branch=z9hG4bKPj5Lf4x.8V7RWblSxehhKV7Ptpu54EFBE8;alias
Server: Twilio
Proxy-Authenticate: Digest realm="sip.twilio.com",qop="auth",nonce="gLqOWcHxSDagvNJan_w8q0uT42Nj5CP3iR6-0XMisk0Luvt5",opaque="94c979d9f104ff1b2c88fac472a07037"
Content-Length: 0
--end msg--

05:38:44.077    pjsua_acc.c  ....IP address change detected for account 0 (122.97.221.234:2138 --> 112.80.12.234:5447). Updating registration (using method 4)
05:38:44.078 sip_auth_clien  ...Digest algorithm is ""
05:38:44.080   pjsua_core.c  ....TX 1024 bytes Request msg REGISTER/cseq=2052 (tdta0xb59666cc) to TCP 54.172.60.1:5060:
REGISTER sip:otiselevator-apac-stage.sip.twilio.com SIP/2.0
Via: SIP/2.0/TCP 112.80.12.234:5447;rport;branch=z9hG4bKPjpqqItKCHbNW7MTFaeJk0JG1G5uC9t-k8;alias
Route: <sip:otiselevator-apac-stage.sip.twilio.com;transport=tcp;lr>
Max-Forwards: 70
From: <sip:11265018@otiselevator-apac-stage.sip.twilio.com>;tag=HExLqlbt2b5gRXzskRGjTgQE4627fhmd
To: <sip:11265018@otiselevator-apac-stage.sip.twilio.com>
Call-ID: 9VesYyJIOdHCI-cZSFAnfh9SyCpMuntq
CSeq: 2052 REGISTER
Supported: outbound, path
Contact: <sip:11265018@112.80.12.234:5447;transport=TCP;ob>
Expires: 600
Allow: PRACK, INVITE, ACK, BYE, CANCEL, UPDATE, INFO, SUBSCRIBE, NOTIFY, REFER, MESSAGE, OPTIONS
Proxy-Authorization: Digest username="11265018", realm="sip.twilio.com", nonce="gLqOWcHxSDagvNJan_w8q0uT42Nj5CP3iR6-0XMisk0Luvt5", uri="sip:otiselevator-apac-stage.sip.twilio.com", response="144b2d05cd4af4c911dede70049d9951", cnonce="N32BPA.WwqWhyQ0QM3a7WL1L3LEiKpM", opaque="94c979d9f104ff1b2c88fac472a07037", qop=auth, nc=00000001
Content-Length:  0

--end msg--
05:38:44.376   pjsua_core.c  .RX 594 bytes Response msg 200/REGISTER/cseq=2052 (rdata0xb5959d7c) from TCP 54.172.60.1:5060:
SIP/2.0 200 OK
CSeq: 2052 REGISTER
Call-ID: 9VesYyJIOdHCI-cZSFAnfh9SyCpMuntq
From: <sip:11265018@otiselevator-apac-stage.sip.twilio.com>;tag=HExLqlbt2b5gRXzskRGjTgQE4627fhmd
To: <sip:11265018@otiselevator-apac-stage.sip.twilio.com>;tag=66434713_c3356d0b_3aa29b63-2379-4883-a57f-fcdb5e071f2d
Via: SIP/2.0/TCP 112.80.12.234:5447;received=112.80.12.234;rport=5447;branch=z9hG4bKPjpqqItKCHbNW7MTFaeJk0JG1G5uC9t-k8;alias
Server: Twilio
Contact: <sip:11265018@112.80.12.234:5447;transport=TCP;ob>;expires=600,<sip:11265018@122.97.221.234:2138;transport=TCP;ob>;expires=4
Content-Length: 0

--end msg--
05:38:44.378    pjsua_acc.c  ....SIP outbound status for acc 0 is not active
05:38:44.378    pjsua_acc.c  ....sip:11265018@otiselevator-apac-stage.sip.twilio.com: registration success, status=200 (OK), will re-register in 600 seconds

tcp_destroy_transport  -- in 
tcp_destroy  -- in 
tcp_on_destroy  -- in 
05:39:16.516 tcpc0xb595bc1c  TCP transport destroyed normally

================tcp connection closed suddenly and try to re-registration 😂=====

05:42:20.059 tcpc0xb5959bcc  TCP connection closed
05:42:20.060            APP  SIP TCP transport is disconnected from 54.172.60.1:5060: End of file (PJ_EEOF) [status=70016]
05:42:20.060    pjsua_acc.c  Disconnected notification for transport tcpc0xb5959bcc
05:42:20.060 sip_transport.  .Transport tcpc0xb5959bcc shutting down, force=0
05:42:20.061    pjsua_acc.c  .Scheduling re-registration retry for acc 0 in 0 seconds..
tcp_destroy_transport  -- in 
tcp_destroy  -- in 
tcp_on_destroy  -- in 
05:42:20.064 tcpc0xb5959bcc  TCP transport destroyed with reason 70016: End of file (PJ_EEOF)
05:42:20.610 tcpc0xb5959bcc  TCP client transport created
05:42:20.611 tcpc0xb5959bcc  TCP transport 10.42.93.8:39836 is connecting to 54.172.60.0:5060...
05:42:20.612    pjsua_acc.c  Acc 0: setting registration..
05:42:20.653 tcpc0xb595bc1c  .TCP client transport created
05:42:20.654 tcpc0xb595bc1c  .TCP transport 10.42.93.8:52329 is connecting to 54.172.60.2:5060...
05:42:20.698   pjsua_core.c  ..TX 757 bytes Request msg REGISTER/cseq=2053 (tdta0xb59666cc) to TCP 54.172.60.0:5060:
REGISTER sip:otiselevator-apac-stage.sip.twilio.com SIP/2.0
Via: SIP/2.0/TCP 10.42.93.8:52329;rport;branch=z9hG4bKPjEyZpChGHXhC74dPzEKHlBTTh9uZ0bKmF;alias
Route: <sip:otiselevator-apac-stage.sip.twilio.com;transport=tcp;lr>
Max-Forwards: 70
From: <sip:11265018@otiselevator-apac-stage.sip.twilio.com>;tag=Igv51coaTAui8Am9TaQYdRxemkiqIQxQ
To: <sip:11265018@otiselevator-apac-stage.sip.twilio.com>
Call-ID: 9VesYyJIOdHCI-cZSFAnfh9SyCpMuntq
CSeq: 2053 REGISTER
Supported: outbound, path
Contact: <sip:11265018@10.42.93.8:39836;transport=TCP;ob>
Contact: <sip:11265018@112.80.12.234:5447;transport=TCP;ob>;expires=0
Expires: 600
Allow: PRACK, INVITE, ACK, BYE, CANCEL, UPDATE, INFO, SUBSCRIBE, NOTIFY, REFER, MESSAGE, OPTIONS
Content-Length:  0


--end msg--
05:42:20.700    pjsua_acc.c  .Acc 0: Registration sent
05:42:35.924 tcpc0xb5959bcc  TCP transport 10.42.93.8:39836 is connected to 54.172.60.0:5060
05:42:35.926            APP  SIP TCP transport is connected to 54.172.60.0:5060
05:42:36.210   pjsua_core.c  .RX 647 bytes Response msg 407/REGISTER/cseq=2053 (rdata0xb5959d7c) from TCP 54.172.60.0:5060:
SIP/2.0 407 Proxy Authentication required
CSeq: 2053 REGISTER
Call-ID: 9VesYyJIOdHCI-cZSFAnfh9SyCpMuntq
From: <sip:11265018@otiselevator-apac-stage.sip.twilio.com>;tag=Igv51coaTAui8Am9TaQYdRxemkiqIQxQ
To: <sip:11265018@otiselevator-apac-stage.sip.twilio.com>;tag=62911568_c3356d0b_92a31bb9-710c-4c0a-b4a6-6d503fbaef8e
Via: SIP/2.0/TCP 10.42.93.8:52329;received=122.97.221.234;rport=18614;branch=z9hG4bKPjEyZpChGHXhC74dPzEKHlBTTh9uZ0bKmF;alias
Server: Twilio
Proxy-Authenticate: Digest realm="sip.twilio.com",qop="auth",nonce="zjWZM8WCNFPcJrh6hsoM7k0FRQfEC1b6J1dZ5_beWjl6-kyw",opaque="db7a6e21cb612cfc3aec3ba1e3b90585"
Content-Length: 0


--end msg--
05:42:36.211    pjsua_acc.c  ....IP address change detected for account 0 (10.42.93.8:39836 --> 122.97.221.234:18614). Updating registration (using method 4)
05:42:36.212 sip_auth_clien  ...Digest algorithm is ""
05:42:36.214   pjsua_core.c  ....TX 1099 bytes Request msg REGISTER/cseq=2054 (tdta0xb59666cc) to TCP 54.172.60.0:5060:
REGISTER sip:otiselevator-apac-stage.sip.twilio.com SIP/2.0
Via: SIP/2.0/TCP 122.97.221.234:18614;rport;branch=z9hG4bKPjfkPnPmwenW2AI8iBIOYoX4Ff3U5Pb7a0;alias
Route: <sip:otiselevator-apac-stage.sip.twilio.com;transport=tcp;lr>
Max-Forwards: 70
From: <sip:11265018@otiselevator-apac-stage.sip.twilio.com>;tag=Igv51coaTAui8Am9TaQYdRxemkiqIQxQ
To: <sip:11265018@otiselevator-apac-stage.sip.twilio.com>
Call-ID: 9VesYyJIOdHCI-cZSFAnfh9SyCpMuntq
CSeq: 2054 REGISTER
Supported: outbound, path
Contact: <sip:11265018@122.97.221.234:18614;transport=TCP;ob>
Contact: <sip:11265018@112.80.12.234:5447;transport=TCP;ob>;expires=0
Expires: 600
Allow: PRACK, INVITE, ACK, BYE, CANCEL, UPDATE, INFO, SUBSCRIBE, NOTIFY, REFER, MESSAGE, OPTIONS
Proxy-Authorization: Digest username="11265018", realm="sip.twilio.com", nonce="zjWZM8WCNFPcJrh6hsoM7k0FRQfEC1b6J1dZ5_beWjl6-kyw", uri="sip:otiselevator-apac-stage.sip.twilio.com", response="52ca9c961a6e484ca2d0c3622ffb4987", cnonce="N32BPA.WwqWhyQ0QM3a7WL1L3LEiKpM", opaque="db7a6e21cb612cfc3aec3ba1e3b90585", qop=auth, nc=00000001
Content-Length:  0


--end msg--
05:42:36.502   pjsua_core.c  .RX 538 bytes Response msg 200/REGISTER/cseq=2054 (rdata0xb5959d7c) from TCP 54.172.60.0:5060:
SIP/2.0 200 OK
CSeq: 2054 REGISTER
Call-ID: 9VesYyJIOdHCI-cZSFAnfh9SyCpMuntq
From: <sip:11265018@otiselevator-apac-stage.sip.twilio.com>;tag=Igv51coaTAui8Am9TaQYdRxemkiqIQxQ
To: <sip:11265018@otiselevator-apac-stage.sip.twilio.com>;tag=16142493_c3356d0b_7ca79680-20e8-4f48-a272-ec7d501073a6
Via: SIP/2.0/TCP 122.97.221.234:18614;received=122.97.221.234;rport=18614;branch=z9hG4bKPjfkPnPmwenW2AI8iBIOYoX4Ff3U5Pb7a0;alias
Server: Twilio
Contact: <sip:11265018@122.97.221.234:18614;transport=TCP;ob>;expires=600
Content-Length: 0


--end msg--
05:42:36.504    pjsua_acc.c  ....SIP outbound status for acc 0 is not active
05:42:36.504    pjsua_acc.c  ....sip:11265018@otiselevator-apac-stage.sip.twilio.com: registration success, status=200 (OK), will re-register in 600 seconds
05:42:52.008 tcpc0xb595bc1c  TCP transport 10.42.93.8:52329 is connected to 54.172.60.2:5060
05:42:52.009            APP  SIP TCP transport is connected to 54.172.60.2:5060
tcp_destroy_transport  -- in 
tcp_destroy  -- in 
tcp_on_destroy  -- in 
05:42:53.658 tcpc0xb595bc1c  TCP transport destroyed normally



=== TCP connection closed again and try to registration😂=====

05:44:41.474 tcpc0xb5959bcc  TCP connection closed
05:44:41.475            APP  SIP TCP transport is disconnected from 54.172.60.0:5060: End of file (PJ_EEOF) [status=70016]
05:44:41.475    pjsua_acc.c  Disconnected notification for transport tcpc0xb5959bcc
05:44:41.475 sip_transport.  .Transport tcpc0xb5959bcc shutting down, force=0
05:44:41.476    pjsua_acc.c  .Scheduling re-registration retry for acc 0 in 3 seconds..
tcp_destroy_transport  -- in 
tcp_destroy  -- in 
tcp_on_destroy  -- in 
05:44:41.479 tcpc0xb5959bcc  TCP transport destroyed with reason 70016: End of file (PJ_EEOF)

05:44:44.906 tcpc0xb5959bcc  TCP client transport created
05:44:44.908 tcpc0xb5959bcc  TCP transport 10.42.93.8:40713 is connecting to 54.172.60.2:5060...
05:44:44.909    pjsua_acc.c  Acc 0: setting registration..
05:44:44.955 tcpc0xb595bc1c  .TCP client transport created
05:44:44.956 tcpc0xb595bc1c  .TCP transport 10.42.93.8:60869 is connecting to 54.172.60.3:5060...
05:44:44.994 tcpc0xb595e94c  ..TCP client transport created
05:44:44.997 tcpc0xb595e94c  ..TCP transport 10.42.93.8:38321 is connecting to 54.172.60.0:5060...
05:44:44.998   pjsua_core.c  ..TX 759 bytes Request msg REGISTER/cseq=2055 (tdta0xb59666cc) to TCP 54.172.60.0:5060:
REGISTER sip:otiselevator-apac-stage.sip.twilio.com SIP/2.0
Via: SIP/2.0/TCP 10.42.93.8:60869;rport;branch=z9hG4bKPjJT-8dx1b0zhcsKwiuGjWIODy1Hv31SUd;alias
Route: <sip:otiselevator-apac-stage.sip.twilio.com;transport=tcp;lr>
Max-Forwards: 70
From: <sip:11265018@otiselevator-apac-stage.sip.twilio.com>;tag=OnIvknP0m1CUrgqZcM79roAs7Rk3CTRE
To: <sip:11265018@otiselevator-apac-stage.sip.twilio.com>
Call-ID: 9VesYyJIOdHCI-cZSFAnfh9SyCpMuntq
CSeq: 2055 REGISTER
Supported: outbound, path
Contact: <sip:11265018@10.42.93.8:40713;transport=TCP;ob>
Contact: <sip:11265018@122.97.221.234:18614;transport=TCP;ob>;expires=0
Expires: 600
Allow: PRACK, INVITE, ACK, BYE, CANCEL, UPDATE, INFO, SUBSCRIBE, NOTIFY, REFER, MESSAGE, OPTIONS
Content-Length:  0


--end msg--
05:44:45.001    pjsua_acc.c  .Acc 0: Registration sent
05:44:45.245 tcpc0xb595e94c  TCP transport 10.42.93.8:38321 is connected to 54.172.60.0:5060
05:44:45.246            APP  SIP TCP transport is connected to 54.172.60.0:5060
05:44:45.515   pjsua_core.c  .RX 646 bytes Response msg 407/REGISTER/cseq=2055 (rdata0xb595eafc) from TCP 54.172.60.0:5060:
SIP/2.0 407 Proxy Authentication required
CSeq: 2055 REGISTER
Call-ID: 9VesYyJIOdHCI-cZSFAnfh9SyCpMuntq
From: <sip:11265018@otiselevator-apac-stage.sip.twilio.com>;tag=OnIvknP0m1CUrgqZcM79roAs7Rk3CTRE
To: <sip:11265018@otiselevator-apac-stage.sip.twilio.com>;tag=57923855_c3356d0b_b486fe0e-90b0-42fb-8edc-c27eba32f9f0
Via: SIP/2.0/TCP 10.42.93.8:60869;received=112.80.12.234;rport=57351;branch=z9hG4bKPjJT-8dx1b0zhcsKwiuGjWIODy1Hv31SUd;alias
Server: Twilio
Proxy-Authenticate: Digest realm="sip.twilio.com",qop="auth",nonce="5HapK5rC79ZqAZLnNUhck0_-rmilo2MudbWGL0Swai8FpQ_H",opaque="9e31749e45b158fa0d1ea75e89a70558"
Content-Length: 0
--end msg--

============assertion here😂======
otis_gateway: ../src/pjsua-lib/pjsua_acc.c:1805: acc_check_nat_addr: Assertion `contact_hdr != ((void *)0)' failed.
```

**Comment by trengginas** [comment]
Please try again using the latest code or 2.14. Let us know if the issue persists. Log level 5 will helpful

**Comment by sauwming** [comment]
I believe this should have been fixed by #3102 . Let us know if this is not the case.

**Comment by ysm0420** [comment]
Thank you , I will try the latest code and update here if the issue persist.
